### ISSUE TITLE
Controlling pan/tilt motorized head

### ISSUE THREAD

**Issue by jumpjack** [falk0069/sony-pm-alt#28]
This product is said to be controllable by playmemories mobile app:

![image](https://user-images.githubusercontent.com/1620953/222372523-ddc411e7-8754-489d-8445-62a932096db0.png)

https://www.amazon.it/-/en/YT-260-Motorized-Camera-Swivel-Control/dp/B07TT33QQ1/ref=pd_lpo_4?pd_rd_w=PAzKv&content-id=amzn1.sym.992f08af-eb2a-4a7b-90a4-cd77032caf05&pf_rd_p=992f08af-eb2a-4a7b-90a4-cd77032caf05&pf_rd_r=Z0192CCABF6GS1HN48MR&pd_rd_wg=eeh7Q&pd_rd_r=a56d929d-87c9-4f25-9196-ed0e9ee538c6&pd_rd_i=B07TT33QQ1&psc=1

> 【Controllo APP e telecomando】 - Insieme all'APP per il controllo remoto della fotocamera nel telefono cellulare, puoi scattare foto e video sul tuo telefono cellulare e la testa panoramica può aiutare la fotocamera a trovare l'oggetto. Software di controllo remoto del telefono cellulare per **YT-26PLAYMEMORIES**. Inclinazione, il telecomando del telefono cellulare può aiutarti a riprendere video o foto per GOPRO entro 20 metri. Dotato di un telecomando separato.

Is the protocol known? Any link or tip?

**Comment by r00t-** [comment]
this seems very off-topic here, this is a project specifically about sony cameras...
also i don't understand how you seem to expect anybody here can read italian.
consider asking your question in a more general forum, e.g. https://www.stackoverflow.com/ .
(have you TRIED if the thing can be controlled by play memories? i doubt that is has a provision for such a thing.)

**Comment by jumpjack** [comment]
> this seems very off-topic here, this is a project specifically about sony cameras...

Why? I wrote "This product is said to be controllable by **playmemories mobile** app", and I linked the page which says that; isn't this repo about replicating Sony Playmemories features?

I **do not**  own this device: I will purchase it only if I can control it by my own app, which could be based on this repo. I guess the author of sony-pm-alt knows very well sony playmemories protocol, which possibly includes pan/tilt features, which of course are not used by the app, but would be useful for me.

So I repeat the question: is the Sony Playmemories protocol known? Any documentation available?

**Comment by falk0069** [comment]
Yeah, this app is not likely to help you.  This app is just an UPNP listener and links the gphoto2 app to the PTP/IP service the camera starts up.  The real magic is with gphoto2 that does the PTP/IP communicating.  You might want to reach out to that community.  But even so the PTP (Picture Transfer Protocol) is geared towards transferring pictures, so probably unlikely it will be able to control movement.

**Comment by r00t-** [comment]
this is what google translate makes of the italian:

> [APP Control & Remote Control】 - Together with the camera remote control APP in the mobile phone, you can take photos and videos on your mobile phone, and the panning head can help the camera find the object. Cell Phone Remote Control Software for YT-26PLAYMEMORIES. Tilt, mobile phone remote control can help you shoot video or photo for GOPRO within 20 meters. Equipped with a separate remote control.

have you looked at the images in the listing:
https://m.media-amazon.com/images/W/IMAGERENDERING_521856-T1/images/I/61IEV1KUafL._AC_SL1001_.jpg

i don't think this pan/tilt mount will have wifi AND a plain rf remote.
(and even less that the remote is wifi.)

i'm pretty sure that, instead, that pan/tilt mount is ONLY controlled by that rf remote,
and they suggest to pair it with an app-controlled camera.

you will have to find a different (and probably more expensive) product for a programmable pan/tilt mount.
(or hack this one and hook it up to an arduino? or build your own from rc servos?)
i actually own an old and very robust electric pan-tilt mount made for SLRs.
the problem is that such devices, made for manual control, won't have any position feedback sensors,
and adding sensors will be rather complex mechanical work, so it's probably easier to build one from scratch.

**Comment by jumpjack** [comment]
Thanks to ChatGPT I discovered some pan/tilt motorized heads (or "gimbals") which are controlled over bluetooth using the  protocol "Feiyutech G", and the AI even provided an example source code to control them and position them in a specific position. 
Amazing, although offtopic here.
But I will ask her/him also something about sony protocols...
